src/catchup/ApplyBucketsWork.cpp

```diff
@@ -107,9 +134,19 @@ ApplyBucketsWork::doReset()
     mLastAppliedSizeMb = 0;
     mLastPos = 0;
     mMinProtocolVersionSeen = UINT32_MAX;
+    mSeenKeys.clear();
+    mBucketsToIndex.clear();
 
     if (!isAborting())
     {
+        if (mOffersOnly)
+        {
+            // The current size of this set is 1.6 million during BucketApply
+            // (as of 12/20/23). There's not a great way to estimate this, so
+            // reserving with some extra wiggle room
+            mSeenKeys.reserve(2'000'000);
```

Comment:
It's pretty variable and correlated to high DEX activity rather than BucketList size. Currently, it takes about 180 mb. I also tested a period of very high DEX activity in 2018 which consumed 1.15 GB. For replaying all history, I would estimate the upper bound is around 1.2 GB (I found an expensive period, but I'm not sure if it's the absolute max).

Reply:
Looks like the peak memory usage of core on any given range is 3.8 GB during catchup (total, not just the offer apply index).